File: youtube_tnt_trigger.js
const {google} = require('googleapis');
const API_KEY = 'YOUR_YOUTUBE_API_KEY';

function extractVideoId(url) {
  const match = url.match(/[?&]v=([\w-]{11})/);
  return match ? match[1] : null;
}

function triggerTnt() {
  console.log('TNT acionada!');
}

async function fetchLikeCount(videoId) {
  const youtube = google.youtube({version: 'v3', auth: API_KEY});
  const res = await youtube.videos.list({
    part: 'statistics',
    id: videoId,
  });
  const items = res.data.items || [];
  if (items.length === 0) {
    throw new Error('Vídeo não encontrado.');
  }
  return parseInt(items[0].statistics.likeCount || '0', 10);
}

async function monitorLikes(videoId) {
  let prevLikes = 0;
  while (true) {
    try {
      const likes = await fetchLikeCount(videoId);
      if (likes > prevLikes) {
        console.log(`Novos likes detectados: ${likes} (anterior ${prevLikes})`);
        triggerTnt();
        prevLikes = likes;
      }
    } catch (err) {
      console.error(err.message);
      return;
    }
    await new Promise(r => setTimeout(r, 10000));
  }
}

async function main() {
  const url = process.argv[2];
  if (!url) {
    console.log('Uso: node youtube_tnt_trigger.js <url_da_live>');
    process.exit(1);
  }
  const videoId = extractVideoId(url);
  if (!videoId) {
    console.log('URL da live inválida.');
    process.exit(1);
  }
  await monitorLikes(videoId);
}

main();

File: youtube_tnt_trigger.py
import re
import sys
import time
from typing import Optional

from googleapiclient.discovery import build

API_KEY = "YOUR_YOUTUBE_API_KEY"


def extract_video_id(url: str) -> Optional[str]:
    """Extract the video ID from a YouTube URL."""
    match = re.search(r"v=([\w-]{11})", url)
    if match:
        return match.group(1)
    return None


def trigger_tnt() -> None:
    """Placeholder for TNT activation."""
    print("TNT acionada!")


def monitor_likes(video_id: str) -> None:
    youtube = build("youtube", "v3", developerKey=API_KEY)
    prev_likes = 0
    while True:
        req = youtube.videos().list(part="statistics", id=video_id)
        resp = req.execute()
        items = resp.get("items", [])
        if not items:
            print("Vídeo não encontrado.")
            return
        stats = items[0]["statistics"]
        likes = int(stats.get("likeCount", 0))
        if likes > prev_likes:
            print(f"Novos likes detectados: {likes} (anterior {prev_likes})")
            trigger_tnt()
            prev_likes = likes
        time.sleep(10)


if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Uso: python youtube_tnt_trigger.py <url_da_live>")
        sys.exit(1)
    url = sys.argv[1]
    video_id = extract_video_id(url)
    if not video_id:
        print("URL da live inválida.")
        sys.exit(1)
    monitor_likes(video_id)

